xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -247,6 +248,25 @@ def extract_var(self, a: ParsingState) -> ConstraintVariableType:
                 types = (types,)
             return self.inner.extract_var(types)
 
+    @dataclass(frozen=True)
+    class _OperandResultInferenceExtractor(VarExtractor[ParsingState]):
+        idx: int
+        is_operand: bool
+        inner: VarExtractor[Sequence[Attribute]]
+        constr: RangeConstraint
```

Comment:
```suggestion
    class _OperandResultInferenceExtractor(VarExtractor[ParsingState]):
        idx: int
        KW_ONLY
        is_operand: bool
        inner: VarExtractor[Sequence[Attribute]]
        constr: RangeConstraint
```

Reply:
should help read the code below

Reply:
I've never seen this before

Reply:
https://docs.python.org/3/library/dataclasses.html#dataclasses.KW_ONLY

Reply:
I misremembered the API

Reply:
I truly hate that API but agree it would make the code better